src/main/java/stsjorbsmod/cards/wanderer/Amnesia.java

```diff
@@ -38,9 +41,19 @@ public Amnesia() {
 
     @Override
     public void use(AbstractPlayer p, AbstractMonster _) {
-        if(dontTriggerOnUseCard) {
-            addToBot(new SnapAction(p));
-            addToBot(new RemovePowersMatchingPredicateAction(p, power -> power.type == AbstractPower.PowerType.BUFF || power.type == AbstractPower.PowerType.DEBUFF));
+        if (dontTriggerOnUseCard) {
+            List<AbstractPower> buffs = p.powers.stream()
+                    .filter(power -> power.type == AbstractPower.PowerType.BUFF)
+                    .collect(Collectors.toList());
+            if (buffs.size() == 0) {
+                AbstractDungeon.actionManager.addToBottom(new TalkAction(true,"~...~", 2.0f, 2.0f));
+            } else if (p.hasPower(ArtifactPower.POWER_ID)) {
+                AbstractDungeon.actionManager.addToBottom(new ReducePowerAction(p, p, ArtifactPower.POWER_ID, 1));
+            } else {
+                int index = AbstractDungeon.miscRng.random(buffs.size());
```

Comment:
I think the argument to random is off-by-one; the range of `random(n)` is 0 through n *inclusive* (they did something different from most other random APIs I've worked with)

Reply:
Yeah, I remember changing what I was doing many times and forgot to bring this back to n-1